core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -638,21 +660,42 @@ class CityInfo {
         demandedResource = ""
     }
 
-    fun reassignPopulation() {
+    // Reassign all Specialists and Unlock all tiles
+    // Mainly for automated cities, Puppets, just captured
+    fun reassignAllPopulation() {
+        manualSpecialists = false
+        reassignPopulation(resetLocked = true)
+    }
+
+    fun reassignPopulation(resetLocked: Boolean = false) {
         var foodWeight = 1f
         var foodPerTurn = 0f
         while (foodWeight < 3 && foodPerTurn <= 0) {
-            workedTiles = hashSetOf()
-            population.specialistAllocations.clear()
-            for (i in 0..population.population)
-                population.autoAssignPopulation(foodWeight)
+            if (resetLocked) {
+                workedTiles = hashSetOf()
+                lockedTiles = hashSetOf()
+            } else {
+                workedTiles = lockedTiles
+            }
+            if (!manualSpecialists)
+                population.specialistAllocations.clear()
+            population.autoAssignPopulation(foodWeight)
             cityStats.update()
 
             foodPerTurn = foodForNextTurn().toFloat()
             foodWeight += 0.5f
         }
     }
 
+    fun isFocus(stat: Stat): Boolean {
```

Comment:
The CityInfo regions are already broken, but no need to place this in "state-changing functions" - could be obsolete with my enum fields suggestion anyway